crosshair/condition_parser.py

```diff
@@ -806,10 +806,89 @@ def get_class_invariants(self, cls: type) -> List[ConditionExpr]:
         return []
 
 
+class GivenParser(ConcreteConditionParser):
+    def __init__(self, toplevel_parser: ConditionParser = None):
+        super().__init__(toplevel_parser)
+
+    @staticmethod
+    def is_string_literal(node: ast.AST) -> bool:
+        if sys.version_info >= (3, 8):
+            return (
+                    isinstance(node, ast.Expr)
+                    and isinstance(node.value, ast.Constant)
+                    and isinstance(node.value.value, str)
+            )
+        else:
+            return isinstance(node, ast.Expr) and isinstance(node.value, ast.Str)
+
+    @staticmethod
+    def check_decorator(fn: Callable) -> bool:
+        # TODO: FIX THIS.
+        source = inspect.getsource(fn)
+        index = source.find("def ")
+        decorators = [
+            line.strip().split()[0]
+            for line in source[:index].strip().splitlines()
+            if line.strip()[0] == "@"
+        ]
+        for decorator in decorators:
+            if "given" in decorator:
+                return True
+        return False
+
+    def get_fn_conditions(self, ctxfn: FunctionInfo) -> Optional[Conditions]:
+        fn_and_sig = ctxfn.get_callable()
+        if fn_and_sig is None:
+            return None
+        (fn, sig) = fn_and_sig
+        # TODO replace this guard with package-level configuration?
+        if (
+                getattr(fn, "__module__", False)
+                and fn.__module__.startswith("crosshair.")
+                and not fn.__module__.endswith("_test")
+        ):
+            return None
+
+        try:
+            decorator = GivenParser.check_decorator(fn)
```

Comment:
As per discussion, this can check for the `fn.hypothesis` attribute, instead of using `check_decorator()` :slightly_smiling_face: